cpp/include/cuml/tree/decisiontree.hpp

```diff
@@ -207,13 +207,15 @@ void decisionTreeClassifierFit(const raft::handle_t &handle,
                                const int ncols, const int nrows, int *labels,
                                unsigned int *rowids, const int n_sampled_rows,
                                int unique_labels,
-                               DecisionTree::DecisionTreeParams tree_params);
+                               DecisionTree::DecisionTreeParams tree_params,
+                               uint64_t seed);
```

Comment:
so, now we will have a `seed` parameter in `RF_params` and in the decisiontree? Any reason for exposing this new argument?

Reply:
We need seed for controlling the randomization needed while training a tree. The `RF_params` is not available in `decisionTreeClassifierFit` or other similar tree functions.

Reply:
How about we introduce a `seed` member inside `DecitionTreeParams`? What do you think?

Reply:
It can be done but I am not sure it is correct idea (it could be). The `DecisionTreeParams` are same for all trees in a forest. Currently, even the `seed` is same for all trees and is `RF_params` seed. Combination of `seed` and `treeid` is used to derive the sub-seed. This is not explicitly done but `seed` and `treeid` are used together. For example [here](https://github.com/vinaydes/cuml/blob/229577ae777e7b6e0c5c02057ed9796ca86f5582/cpp/src/decisiontree/batched-levelalgo/kernels.cuh#L361). I am not sure we could always take this approach for other places where randomization would be needed. For example, for tie-breaks in choosing splits.
I am alright with moving `seed` to `DecisionTreeParams` but we might have to take it out in future, if we need different seeds for different trees.

Reply:
Good point. Let's keep your current change as-is.